### PR TITLE
GPU instance flag + README updated

### PR DESCRIPTION
**Pull request by CHIROOTH**
Gpu instances can now be accessed using kslurm gpu flag, can specify the type and the number of instances wanted for jobs. 
Re-wrote the README

### CODE REVIEW

**Comment by akhanf** [comment]
Hi Chirooth,

Are you able to add some test cases for this too? Should be able to follow the same structure as the existing tests..

File: kslurm/test/args/test_gpu_args.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        ("gpu=foo:0", "foo:1"),
+        ("gpu=bar:999", "bar:999"),
+    ],
+)
+def test_gpu_formatter(raw, expected):
+    assert fmt(raw) == expected
+
+
+# check for edge cases :
+# invalid input ("gpu=foo:-1", "gpu=bad:abc", "gpu=", non-strings)
```
Comment (Dhananjhay) [inline kslurm/test/args/test_gpu_args.py:24]: Do we handle "gpu=" case?

File: kslurm/test/test_gpu_flag.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        out = capsys.readouterr().out
+
+        normalized = " ".join(out.split())
+        assert (
+            "--account=some-account --time=2-09:11:00 --cpus-per-task=8 "
+            "--mem=5000 --gpus-per-node=h100:2"
+        ) in normalized
+        subprocess.assert_called_with(
+            "echo '#!/bin/bash\n\n./kslurm command' | sbatch --account=some-account "
```
Comment (Dhananjhay) [inline kslurm/test/test_gpu_flag.py:62]: I had to specify ./kslurm here to get the test passing, but is that the expected behavior?

File: kslurm/test/test_gpu_flag.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            "--time=2-09:11:00 --cpus-per-task=8 --mem=5000 --gpus-per-node=h100:2 "
+            "--parsable ",
+            shell=True,
+            stdout=-1,
+            stderr=-2,
+        )
+        assert Path.cwd() == starting_cwd
```
Comment (Dhananjhay) [inline kslurm/test/test_gpu_flag.py:69]: Same here, I had to remove `/ "kslurm"` to get the test passing.
